Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.cpp

```diff
@@ -296,8 +296,12 @@ InstantiationResult AbstractMachine::instantiate(Module const& module, Vector<Ex
             auto current_index = index;
             ++index;
             auto active_ptr = segment.mode.get_pointer<ElementSection::Active>();
-            if (!active_ptr)
+            auto elem_instance = m_store.get(main_module_instance.elements()[current_index]);
+            if (!active_ptr) {
+                if (segment.mode.has<ElementSection::Declarative>())
+                    *m_store.get(main_module_instance.elements()[current_index]) = ElementInstance(elem_instance->type(), {});
```

Comment:
you could just use `*elem_instance = ElementInstance(elem_instance->type(), {});` here